src/argilla/client/sdk/datasets/api.py

```diff
@@ -35,11 +36,40 @@ def get_dataset(client: AuthenticatedClient, name: str) -> Response[Dataset]:
         timeout=client.get_timeout(),
     )
 
-    return _build_response(response=response, name=name)
+    if response.status_code == 200:
+        response_obj = Response.from_httpx_response(response)
+        response_obj.parsed = Dataset(**response.json())
+        return response_obj
+    return handle_response_error(response)
```

Comment:
Indeed, this line will raise an error instead of returning something. 
```suggestion
    handle_response_error(response)
```